### DISCUSSION TITLE
Run 2 or more containers with GUI Firefox on a single PC

### DISCUSSION THREAD

**Discussion by Andrewgmiles** [jlesage/docker-firefox#237] [General]
One container on 5800:5800 runs just fine, however if i setup more firefox containers ie. 5801 5802 they create in docker but Firefox does not start

Is there a way to open more ports?
I want to run 200 containers all using Firefox on a single machine

**Comment by jlesage** [comment]
This should work. Are you using the default bridge network ?

**Reply by seoetraffic** [reply]
can you explain more how to install more than 1 firefox in the same docker? I use ubuntu VPS

**Comment by B-Down** [comment]
In order to be able to run more than one concurrent occurences of the same image **and/or** different tools that use the same port(s) (since web and vnc services usually always use the same internal ports):

- Make sure you increment only the external port(s) you're opening, as the internal one has a default value. For instance:
  - Web:
    - First container: 5800:5800
    - Second container: 5801:5800
    - Third container: 5802:5800
  - Same thing for VNC:
    - First: 5900:5900
    - Second: 5901:5900
    - Third: 5902:5900
  - So on and so forth...
- Create a new directory for each container:
  - First container: **Host/Volume** = /docker/appdata/firefox-1 ; **Path in container** = /config
  - Second container: **Host/Volume** = /docker/appdata/firefox-2 ; **Path in container** = /config
  - Third container: **Host/Volume** = /docker/appdata/firefox-3 ; **Path in container** = /config

Unless you're dead-set on using the terminal for your Docker deployments, you may want to have a look at Portainer, as it significantly simplifies dealing with Docker *stuff*.

**Reply by B-Down** [reply]
@Andrewgmiles @seoetraffic
